packages/flutter/lib/src/widgets/container.dart

```diff
@@ -102,6 +110,10 @@ class Container extends StatelessWidget {
     assert(padding == null || padding.isNonNegative);
     assert(decoration == null || decoration.debugAssertIsValid());
     assert(constraints == null || constraints.debugAssertIsValid());
+    assert(color == null || decoration == null,
```

Comment:
The problem is that `decoration` has the type `Decoration` rather than `BoxDecoration`.  That means it can be an arbitrary custom decoration (like the `FlutterLogoDecoration`) and might not have the concept of color.